docs/learn/learn-staking-advanced.md

```diff
@@ -111,62 +111,75 @@ On Polkadot and Kusama, the instance of the pallet
-making it extremely scalable. Let us explore the sorting functionality of the bags list with an
-example. In the bags list below, there are 8 nodes (corresponding to 8 accounts with staked funds)
-placed in 3 bags. It can be observed that the list of nodes within the bags is arranged based on
-their insertion order and not based on the number of tokens bonded. For instance, the nodes in bag 1
-are arranged in this order: 15 → 12 → 19.
-
-![bags list example 1](../assets/staking/bags-list-example-1.png)
-
-Let's say the nominator with the stake of 19 DOT bonds 2 DOT additionally. This action would place
-that nominator node in bag 2, right after the node with 27 DOT.
-
-![bags list example 2](../assets/staking/bags-list-example-2.png)
-
-Once the nomination period ends, the NPoS election mechanism takes the nomination intents and their
-associated votes as input, and it outputs a set of validators. The bags are iterated from the most
-staked to the least staked. This could leave the last touched bag to only be partially iterated.
-This means that in some edge cases, the order of members within a bag is also important. Recall that
-within each bag, the iteration order is simply the insertion order. If only 7 nodes must be picked
-for the electing set, the nodes with 5 and 7 DOT will be selected while the node with 8 DOT will be
-left out.
-
-![bags list example 3](../assets/staking/bags-list-example-3.png)
-
-If one receives staking rewards and the amount of staked tokens within the stash account increases
-over time, the position within a bag changes and may also result in a change of bag. This may also
-happen if accounts within the bag bond more tokens or unbond tokens, one's account position and the
-position of other accounts in the bags list might change. These changes are not done automatically,
-requiring the nominator to submit the permissionless extrinsic `rebag` within the `voterList` pallet
-instance to update their position. This allows anyone to specify another account that is in the
-wrong bag, and place it in the correct one. The `voterList` pallet instance also comes with the
-extrinsic `putInFrontOf` which helps the node to move up in the bag. Actions like bonding/unbonding
-tokens automatically rebags the nominator node, but events like staking rewards/slashing do not. See
-the [bags-list](learn-nominator.md#bags-list) section for more information.
-
-This sorting functionality using bags is extremely important for the
+In {{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} NPoS nomination intents are
+sorted in a list called [bags-list](https://github.com/paritytech/substrate/pull/9507).
+{{ kusama: The bags list example below uses DOT for explaining the concepts. :kusama }}The Bags-List
+substrate pallet is designed to be self-maintaining, with minimal effort from the blockchain, making
+it extremely scalable. This sorting functionality using bags is extremely important for the
```

Comment:
"This sorting functionality..." - we have not introduced the sorting functionality used in the bags list yet. It makes sense to move this line and the reference to the long-term improvements at the end of the section.